### PR TITLE
Attach an acl policy to created namespaces

### PR DESCRIPTION
**Pull request by adilyse**
This is necessary if ACLs are enabled to grant cross Consul namespace
communication.

### CODE REVIEW

File: subcommand/sync-catalog/command.go [hunk 1]
```diff
@@ -123,6 +124,9 @@ func (c *Command) init() {
 		"namespace mirroring")
 	c.flags.StringVar(&c.flagK8SNSMirroringPrefix, "k8s-namespace-mirroring-prefix", "",
 		"[Enterprise Only] Prefix that will be added to all k8s namespaces mirrored into Consul if mirroring is enabled.")
+	c.flags.StringVar(&c.flagCrossNamespaceACLPolicy, "cross-consul-namespace-acl-policy", "",
```
Comment (ishustava) [inline subcommand/sync-catalog/command.go]: Would it make more sense to name it `consul-cross-namespace-acl-policy` to match other names?

**Comment by adilyse** [comment]
These methods are functionally to the one in the acl bootstrapper that is tested extensively. For the sync one, it doesn't make sense to add tests before the rest of the tests are merged. For the injector, this requires a bunch of additional test setup that we felt okay skipping when we chatted earlier today.

All doable, but a lot of extra work.

File: subcommand/sync-catalog/command.go (hunk 1 above)
Comment (adilyse) [reply subcommand/sync-catalog/command.go]: I was trying to specify that it was for crossing consul namespaces, but it does end up being a little awkward. I'll update it.

**Comment by adilyse** [comment]
Tests for the connect injector have been added. The sync tests will be added once the sync tests have been merged.

**Pull request merged**